core/src/compensationAuction.ts

```diff
@@ -0,0 +1,185 @@
+    const bidEndDate = auctionData.tic ? new Date(auctionData.tic * 1000) : undefined;
+    const earliestEndDate = bidEndDate ? getEarliestDate(auctionEndDate, bidEndDate) : auctionEndDate;
+    const state = await getAuctionState(network, earliestEndDate, auctionData.tic);
+
+    return {
+        ...baseAuctionInfo,
+        earliestEndDate,
+        bidAmountMKR: new BigNumber(auctionData.bid._hex).div(WAD),
+        receiveAmountDAI: new BigNumber(auctionData.lot._hex).div(RAD),
+        receiverAddress: auctionData.guy,
+        auctionEndDate,
+        bidEndDate,
+        state,
+        fetchedAt,
+        auctionStartDate,
+    };
+};
+
+export const getActiveCompensationAuctionOrUndefined = async (
+    network: string,
+    auctionIndex: number,
+    contractName: 'MCD_FLAP' | 'MCD_FLOP'
+) => {
+    const auction = await fetchCompensationAuctionByIndex(network, auctionIndex, contractName);
+    if (auction.state === 'collected') {
+        return;
+    }
+    return auction;
+};
+
+export const restartCompensationAuction = async function (
+    network: string,
+    auctionIndex: number,
+    contractName: 'MCD_FLOP' | 'MCD_FLAP',
+    notifier?: Notifier
+): Promise<void> {
+    await executeTransaction(network, contractName, 'tick', [auctionIndex], { notifier });
+};
+
+export const getMarketPriceMkr = async function (network: string, mkrAmount: BigNumber): Promise<BigNumber> {
+    try {
+        return new BigNumber(1).div((await convertMkrToDai(network, mkrAmount)).div(mkrAmount));
+    } catch (error) {
+        return new BigNumber(NaN);
+    }
+};
+
+export const getMarketPriceDaiToMkr = async function (network: string, mkrAmount: BigNumber): Promise<BigNumber> {
+    try {
+        return (await convertMkrToDai(network, mkrAmount)).div(mkrAmount);
+    } catch (error) {
+        return new BigNumber(NaN);
+    }
+};
+
+export const collectCompensationAuction = async function (
+    network: string,
+    auctionIndex: number,
+    contractName: 'MCD_FLAP' | 'MCD_FLOP',
+    notifier?: Notifier
+) {
+    const auction = await getActiveCompensationAuctionOrUndefined(network, auctionIndex, contractName);
+    if (!auction || auction.state !== 'ready-for-collection') {
+        throw new Error('Did not find the auction to collect.');
+    }
+    await executeTransaction(network, contractName, 'deal', [auctionIndex], { notifier });
+};
+
+export const getCompensationAuctionTransactionFees = async function (
+    network: string
+): Promise<CompensationAuctionTransactionFees> {
+    const gasPrice = await getGasPriceForUI(network);
+    const exchangeRate = await getMarketPrice(network, 'ETH');
+
+    const restartTransactionFeeEth = gasPrice.multipliedBy(80563);
+    const allowanceTransactionFeeEth = gasPrice.multipliedBy(48373);
+    const bidTransactionFeeEth = gasPrice.multipliedBy(85181);
+    const collectTransactionFeeEth = gasPrice.multipliedBy(94114);
+    const authTransactionFeeEth = gasPrice.multipliedBy(48356);
```

Comment:
Since those are two different contracts that does different things underneath, I wouldn't expect that restarting, bidding and collecting will cost the same amount of gas

Reply:
I beleive i might be lacking info on how to extract the correct info then.

Two factors:

  - if the debt auction never happened - can we still estimate the fees?
  - i'm not familiar with the flow on how to extract them (never done before) and in https://github.com/sidestream-tech/unified-auctions-ui/issues/350 the fees are already listed.

Reply:
It's simple, you need to try executing this transaction on the frontend, and then check it's `gas limit` or calling https://docs.ethers.io/v5/api/contract/contract/#contract-estimateGas with the same parameters. You're right, since there were never auctions of this type estimation will only be possible within a simulation/test

Reply:
So i assume we're blocked here until the simulation is solved.